docs/source/newsfragments/3983.feature.rst

```diff
@@ -0,0 +1 @@
+Added :class:`GpiClock`, a C++ implementation of a simple clock, exposed as :class:`~cocotb.simulator.GpiClock`. It has the advantage of higher performance due to not having to wake up a python task on a Timer to then enqueue a write. The existing :class:`~cocotb.clock.Clock` uses it automatically when it would behave identically to the python implementation.
```

Comment:
```suggestion
Added :class:`cocotb.simulator.GpiClock`, a C++ clock generator implementation with higher performance due to less handshaking between Python and the GPI. :class:`~cocotb.clock.Clock` uses it automatically when it would behave identically to the Python implementation.
```